pexpect/spawnbase.py

```diff
@@ -8,7 +8,11 @@
 from .expect import Expecter, searcher_string, searcher_re
 
 PY3 = (sys.version_info[0] >= 3)
-text_type = str if PY3 else unicode
+try:
+    text_type = unicode
+except NameError:
+    text_type = str
```

Comment:
I actually prefer the explicit PY3 check here. It's frustrating if we have to make the code worse in my opinion just to satisfy an automated tool. As a compromise, we could add a dependency on `six` and just use `six.text_type`.